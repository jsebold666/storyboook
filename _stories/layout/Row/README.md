A `<Row>` component is meant to hold one or more `<Column>` components.
