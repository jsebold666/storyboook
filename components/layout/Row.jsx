import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const Row = ({ className, children, ...props }) => (
  <div className={cx('gds-layout__row', className)} {...props}>
    {children}
  </div>
);

Row.displayName = 'Row';

Row.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string
};

export default Row;
