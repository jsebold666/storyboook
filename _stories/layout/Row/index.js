import React from 'react';
import { object, text } from '@storybook/addon-knobs';

import readme from './README.md';
import Row from '../../../components/layout/Row';

const component = () => (
  <Row className={text('ClassName', '')} style={object('style', { border: '1px dashed red' })}>
    Example
  </Row>
);

export default [readme, component];
