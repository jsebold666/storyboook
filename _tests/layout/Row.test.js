/* globals mount */
import React from 'react';
import Row from '../../components/layout/Row';

describe('Expect <Row>', () => {
  it('to render', () => {
    const props = {
      className: 'cool-stuff'
    };
    const wrapper = mount(
      <Row {...props}>
        <div>Content</div>
      </Row>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
