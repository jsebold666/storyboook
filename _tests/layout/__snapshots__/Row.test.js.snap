// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <Row> to render 1`] = `
<Row
  className="cool-stuff"
>
  <div
    className="gds-layout__row cool-stuff"
  >
    <div>
      Content
    </div>
  </div>
</Row>
`;
